src/common/components/content/widgets/ProfileBox.jsx

```diff
@@ -143,28 +148,55 @@ export default class ProfileBox extends Component {
                     if (this.state.password !== '') {
                         updatedUser.password = this.state.password;
                     }
-                    if (editable) {
+                    if (editable && !this.props.isCurrentUser) {
```

Comment:
This is on purpose for siteAdmins too - they shouldn't be able to reject there site adminship accidentally.